src/plugins/add-cal/add-calendar.js

```diff
@@ -81,13 +81,13 @@ var componentName = "wb-addcal",
 					event_details.description = prop_cache.textContent.replace( /(\r\n|\n|\r)/gm, " " );
 					break;
 				case "startDate":
-					event_details.sDate = dtToISOString( $( "time[property='startDate']" ) );
+					event_details.sDate = dtToISOString( $( "time[property='startDate']", $elm ) );
 					break;
 				case "endDate":
-					event_details.eDate = dtToISOString( $( "time[property='endDate']" ) );
+					event_details.eDate = dtToISOString( $( "time[property='endDate']", $elm ) );
 					break;
 				case "location":
-					if ( !prop_cache.getAttribute( "typeof" ) ) {
+					if ( ( !prop_cache.getAttribute( "typeof" ) !== "Place" ) && ( $( prop_cache ).find( "[property=url]" ) ) ) {
```

Comment:
I think the leading `!` is messing up the readability and maybe even the efficiency of this conditional. Please review and also I think it would be relevant to have a mention of `VirtualLocation` somewhere in it, just like `Place`.

Reply:
Revised the condition. Should be much clearer now.